oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/ClientSecretAuthenticationProvider.java

```diff
@@ -93,6 +95,11 @@ public Authentication authenticate(Authentication authentication) throws Authent
 			throwInvalidClient(OAuth2ParameterNames.CLIENT_ID);
 		}
 
+		Instant expiredAt = registeredClient.getClientSecretExpiresAt();
+		if (expiredAt!=null && Instant.now().isAfter(expiredAt)) {
```

Comment:
Add spaces between the comparison `expiredAt!=null` so it's consistent with other code